sqlitecluster/SQLite.cpp

```diff
@@ -819,9 +821,27 @@ int SQLite::commit(const string& description, function<void()>* preCheckpointCal
             if (_sharedData.outstandingFramesToCheckpoint) {
                 auto start = STimeNow();
                 int framesCheckpointed = 0;
-                sqlite3_wal_checkpoint_v2(_db, 0, SQLITE_CHECKPOINT_PASSIVE, NULL, &framesCheckpointed);
+
+                // We default to PASSIVE checkpoint everywhere as that has been the value proven to work fine for many years.
+                if (_checkpointMode != "PASSIVE") {
+                    int checkpointMode = SQLITE_CHECKPOINT_PASSIVE;
+                    if (_checkpointMode == "FULL") {
+                        checkpointMode = SQLITE_CHECKPOINT_FULL;
+                    } else if (_checkpointMode == "RESTART") {
+                        checkpointMode = SQLITE_CHECKPOINT_RESTART;
+                    } else if (_checkpointMode == "TRUNCATE") {
+                        checkpointMode = SQLITE_CHECKPOINT_TRUNCATE;
+                    }
+                    // For non-passive checkpoints, we must set a busy timeout in order to wait on any readers.
+                    // We set it to 2 minutes as the majority of transactions should take less than that.
+                    sqlite3_busy_timeout(_db, 120'000);
+                    sqlite3_wal_checkpoint_v2(_db, 0, checkpointMode, NULL, &framesCheckpointed);
+                    sqlite3_busy_timeout(_db, 0);
```

Comment:
This is not supposed to be here, right?

Reply:
That was Dan's recommended code in the email. Do you think we should not have this? Is it because it seems useless?